Dumbdown the components

msafi opened the pull request:
This PR makes framework components delegate to `vanillaTextMask.js`. This greatly simplifies the components but takes away the advantage of cross browser compatibility that the frameworks provide. However, I believe Text Mask should still work on all the supported browsers because it only relies on `input` event, which is widely supported.

I also removed most of the component tests because, again, they now just delegate. I will later need to figure out how to test that this delegation happens correctly.

@browniefed Please take a quick look when you get a chance.

browniefed commented on react/src/reactTextMask.jsx:
```diff
@@ -1,112 +1,35 @@
   }
 })
 
 export default MaskedInput
```
The changes to this file are going to cause the issue we discussed last night where the controlled input value will change but never be re-conformed.

I know you want to bring everything to just be vanilla but you'll have to use the life cycle methods or else this could cause a lot of problems with inputs.

browniefed commented on vanilla/src/vanillaTextMask.js:
```diff
@@ -38,14 +28,27 @@ export function maskInput({
+
+  inputElement.value = state.conformedInput
+
+  inputElement.addEventListener('input', onInput)
```
Using just input will remove IE9 compatibility

msafi commented:
@browniefed I updated the code.

Here's what I changed:

1. I made the `update` method of Vanilla accept an optional `value` argument. It will compare `value` to the previously conformed input. If they are the same, it will halt. If they are different, it will conform `value`. If `value` is not passed, it will conform `inputElement.value`. And now I call `update(value)` in `componentDidUpdate`. There's no flash and it seems to work.
2. I put back event detection in the hands of the framework components. For example, In React's `onChange`, I call `this.control.update()`.

So hopefully this implementation will give us a lot of simplicity without sacrificing the benefits of using framework components.

msafi commented:
I'm gonna merge this PR since I need it for another PR I'm about to submit. @browniefed Let's discuss any issues if you see any later.
